### PR TITLE
Implemented a section regarding Implicit layers.

### PR DESCRIPTION
**Pull request by vitaondr**
I got an extra assignment from Karel Zimmermann. This is a commit containing one .md file with text about Implicit layers. I used the backpropagation section as a template.

### CODE REVIEW

File: docs/implicitLayers.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+1. TOC
+{:toc}
+</details>
+
+## Introduction
+
+Implicit layers are part of neural networks in the same way as any other layer. The difference lies in what these layers compute. For example, a sigmoid layer computes the output variable simply by applying a formula to the input variable. Implicit layers, however, rely on all variables involved in the equation.
+
+{: .note }
+>The explicit function might look something like this.
```
Comment (aleskucera) [inline docs/implicitLayers.md]: I would expect a definition and proper example here, not a note that vaguely describes the implicit function.

File: docs/implicitLayers.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+>The explicit function might look something like this. 
+>$$ y = g(x).$$ 
+>In contrast, an implicit layer is defined by an equation such as:
+>$$ f(x, \omega) = 0 $$
+>The input for the function is $$ \omega $$ and the output optimal value $$ \boldsymbol{x}^{\star}$$.
+
+
+## Types of implicit layers
+
+In this article i will cover these types of implicit layers, as those are covered in the lectures.
```
Comment (aleskucera) [inline docs/implicitLayers.md]: Do not write in first person, keep the text neutral.

File: docs/implicitLayers.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+- **Root finder**
+- **Unconstrained optimizer**
+- **Constrained optimizer**
+- **ODE solver** 
+
+## Backpropagation
+
+Like any other layer, implicit layers must support gradient backpropagation for the network to learn. To achieve this, we rely on implicit differentiation. If the layer’s output $$\boldsymbol{\omega}$$ satisfies an equation of the form:
```
Comment (aleskucera) [inline docs/implicitLayers.md]: I would start that each implicit layer has its own solver (for example, iterative solvers), and it is hard to propagate gradient through them. It is critical to explain to the students the purpose of why you talk here about backpropagation.

File: docs/implicitLayers.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+$$ f(\boldsymbol{x}, \omega) = 0 $$
+
+Where x represents the input, we than calculate the gradient of the loss function with respect to x and the model's parameters as: 
+
+$$
```
Comment (aleskucera) [inline docs/implicitLayers.md]: The purpose of this text is to explain to the students the 'hard' part of the lecture. This is precisely that part. I would give you points just to explain this equation, but that is missing here.

**Review by aleskucera — CHANGES_REQUESTED** [review]
It seems like you attempted to cover too much material for just two points, which resulted in a text that doesn’t effectively explain the topic to students. This is not your fault - it’s likely our oversight, as the topic itself is rather challenging. I recommend providing an explanation of the first 16 slides of this presentation:

[Implicit Layers Presentation](https://cw.fel.cvut.cz/wiki/_media/courses/b3b33urob/lectures/11b_implicit_layers.pdf)

Start with a clear and proper definition of implicit functions and their use cases in neural networks. This will help students understand the importance of these functions and prevent the topic from feeling too vague. Then, explain the gradient of the root finder presented on slide 14. Finally, elaborate on the example given on slide 16. 

If you would need any additional guidance, feel free to contact me.
